go-controller/pkg/node/management-port-dpu.go

```diff
@@ -47,15 +40,10 @@ func (mp *managementPortRepresentor) Create(nodeAnnotator kube.Annotator, waiter
 	// Get management port representor netdevice
 	link, err := util.GetNetLinkOps().LinkByName(mp.repName)
 	if err != nil {
-		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
-			return nil, fmt.Errorf("failed to lookup %s link: %v", mp.repName, err)
-		}
-		// It may fail in case this is not the first run after reboot and management port has already been renamed.
-		link, err = util.GetNetLinkOps().LinkByName(k8sMgmtIntfName)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get link device for %s. %v", mp.repName, err)
-		}
-	} else if mp.repName != k8sMgmtIntfName {
+		return nil, err
+	}
+
+	if link.Attrs().Name != mp.repName {
```

Comment:
Had I missed this last time? It is incorrect, change to `link.Attrs().Name != k8sMgmtIntfName` please